src/client/features/pathways/pathways-cy.js

```diff
@@ -95,6 +95,66 @@ let layout = cy => {
 
 let bindCyEvents = cy => {
 
+  /**
+   * @description Apply style modifications after 200ms delay on `mouseover` for non-compartment nodes.
+   * Currently puts opacity of hovered node & neighbourhood to 1, everything else to 0.3
+   */
+  const nodeHoverMouseOver = _.debounce(evt => {
+    const node = evt.target;
+    const ecAPI = cy.expandCollapse('get');
+    let elesToHighlight = new Set();
+
+    //If node has children and is expanded, do not highlight
+    if (node.isParent() && ecAPI.isCollapsible(node)) { return; }
+
+    //Create a list of the hovered node & its neighbourhood
+    node.neighborhood().nodes().union(node).forEach(node => {
+      node.ancestors().forEach(ancestor => elesToHighlight.add(ancestor));
+      node.descendants().forEach(descendant => elesToHighlight.add(descendant));
+      elesToHighlight.add(node);
+    });
+    node.neighborhood().edges().forEach(edge => elesToHighlight.add(edge));
+
+    //Add highlighted class to node & its neighbourhood, unhighlighted to everything else
+    //batch for improved perf
+    cy.batch( () => {
+      cy.elements().addClass('unhighlighted');
+      elesToHighlight.forEach(ele => {
+        ele.removeClass('unhighlighted');
+        ele.addClass('highlighted');
+      });
+    });
+
+  },200,{leading:false,trailing:true});
+
+    /**
+   * @description Apply style modifications after 200ms delay on `mouseover` for edges.
+   * Currently puts opacity of hovered edge & neighbourhood to 1, everything else to 0.3
+   */
+  const edgeHoverMouseOver = _.debounce(evt => {
+    const edge = evt.target;
+    let elesToHighlight = new Set();
```

Comment:
It's generally better to use a Collection as a set for cy eles.  E.g.

```
let eles = cy.collection();

eles.merge(elesToAdd);
eles.merge(moreElesToAdd);
```

Using a Set could give weird results if you get an Element versus a Collection.